arch/X86/X86IntelInstPrinter.c

```diff
@@ -562,10 +772,24 @@ static void printPCRelImm(MCInst *MI, unsigned OpNo, SStream *O)
 			imm = imm & 0xffff;
 
 		if (imm < 0) {
-			SStream_concat(O, "0x%"PRIx64, imm);
+			switch (MI->csh->syntax) {
+				case CS_OPT_SYNTAX_NASM:
+					SStream_concat(O , "0%"PRIx64"h" , imm);
```

Comment:
This difference seems superfluous:
1. NASM can handle `0x...` notation easily
2. `0...h` is not nice when the most significant digit isn't a letter, like `12345678h`: your code will render it as `012345678h`, where `0` is completely needless and confusing (9 digits instead of 8).

Reply:
If we need to copy and compile it,It's going to be very important.
For example 1.asm:
```
mov rax, 0fffffffh
```
Compiled successfully with command ```nasm -fwin64 1.asm```

if use ```mov rax, fffffffh```,it Will be reported wrong :```error: symbol `fffffffh' undefined```

Reply:
Yes, I agree the leading `0` is necessary if the number starts with a letter, but printing it for all numbers is not nice. Especially so since NASM supports C-style hex number syntax. Why not just leave `0x...` notation as in other syntaxes?

Reply:
This is my carelessness.
and It's my habit. I can't remember that there's no difference between them.- -!